# Accquire

**imports**

In [1]:
import pandas as pd

# system manipulation
import sys
sys.path.append("./util_")
import acquire_
# import prepare_

# other
import warnings
warnings.filterwarnings("ignore")

## Get data

Data from [catalog.data.gov](https://catalog.data.gov/dataset/dohmh-dog-bite-data)

|Column Name|Description|
|----|----|
|**UniqueID**|Unique dog bite case identifier|
|**DateOfBite**|Date bitten|
|**Species**|Animal Type (Dog)|
|**Breed**|Breed type|
|**Age**|Dog's age at time of bite. Numbers with 'M' indicate months.|
|**Gender**|Sex of Dog. M=Male, F=Female, U=Unknown|
|**SpayNeuter**|Surgical removal of dog's reproductive organs. True (reported to DOHMH as Spayed or Neutered), False (Unknown or Not Spayed or Neutered)|
|**Borough**|Dog bite Borough. Other' indicates that the bite took place outside New York City|
|**ZipCode**|Dog bite Zipcode. Blank ZipCode indicates that information was not available|

In [2]:
# get data
dogs = pd.read_csv("https://data.cityofnewyork.us/api/views/rsgh-akpg/rows.csv", index_col=0)

# look at my data
dogs.head()

DateOfBite Species        Breed  Age Gender  SpayNeuter  \
UniqueID                                                                 
1         January 01 2018     DOG      UNKNOWN  NaN      U       False   
2         January 04 2018     DOG      UNKNOWN  NaN      U       False   
3         January 06 2018     DOG     Pit Bull  NaN      U       False   
4         January 08 2018     DOG  Mixed/Other    4      M       False   
5         January 09 2018     DOG     Pit Bull  NaN      U       False   

           Borough ZipCode  
UniqueID                    
1         Brooklyn   11220  
2         Brooklyn     NaN  
3         Brooklyn   11224  
4         Brooklyn   11231  
5         Brooklyn   11224

## Understand the data

In [4]:
dogs.shape

(22663, 8)

In [5]:
dogs.columns

Index(['DateOfBite', 'Species', 'Breed', 'Age', 'Gender', 'SpayNeuter',
       'Borough', 'ZipCode'],
      dtype='object')

In [6]:
dogs.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 22663 entries, 1 to 10280
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   DateOfBite  22663 non-null  object
 1   Species     22663 non-null  object
 2   Breed       20445 non-null  object
 3   Age         11442 non-null  object
 4   Gender      22663 non-null  object
 5   SpayNeuter  22663 non-null  bool  
 6   Borough     22663 non-null  object
 7   ZipCode     16805 non-null  object
dtypes: bool(1), object(7)
memory usage: 1.4+ MB


In [7]:
# count of numeric columns and object columns
numerics = len(dogs.select_dtypes("number").columns)
objects = len(dogs.select_dtypes("object").columns)

print("Numeric col count:", numerics)
print("object col count:", objects)

Numeric col count: 0
object col count: 7


In [8]:
dogs.describe()

DateOfBite Species     Breed    Age Gender SpayNeuter Borough  \
count               22663   22663     20445  11442  22663      22663   22663   
unique               2555       1      1651    228      3          2       6   
top     September 16 2017     DOG  Pit Bull      2      U      False  Queens   
freq                   25   22663      4004   1624  10535      16787    5773   

       ZipCode  
count    16805  
unique     519  
top      10029  
freq       369

**Take aways:**

- I have 22663 rows and 9 columns
- 8 columns are objects and 1 1 boolean
- 0 null count
- I also have descriptive statistics of my data